### ISSUE TITLE
Touchbar support

### ISSUE THREAD

**Issue by pelim** [koekeishiya/chunkwm#120]
Support for Mac Book Pro Touchbar

**Comment by dominiklohmann** [comment]
You actually cannot do that without blocking the touch bar for everything else like BetterTouchTool does. And there's no need to implement that in chunkwm (imo) if you can just use BetterTouchTool and make it execute commands for whatever you want.

You might want to read the [Human Interface Guidelines for macOS Apps]( https://developer.apple.com/macos/human-interface-guidelines/overview/themes/) on this topic, it shows you why background applications cannot and should not support the touch bar.

**Comment by koekeishiya** [comment]
I have no interest in the new MBP Touchbar and will not do any research or otherwise look into how that would work. I'm sure it wouldn't be hard to do if someone is interested in doing so, but I don't really see the point of this being integrated with chunkwm, as it responds to external commands through its dedicated socket.

**Comment by dominiklohmann** [comment]
For what it's worth, I just tried mapping some touch bar buttons with BTT to chunkc commands and while it works, it feels really clunky, unnecessary, slow and worst of all, it does not feel native to macOS at all.

To quote the Human Interface Guidelines I linked above, "[the touch bar] shouldn’t include controls that replicate key-based navigation". 

@koekeishiya The laptops are actually not that bad, just gotta get used to mapping caps lock to esc and not ever resting your fingers on the function row. Battery life's excellent, weight is low and it's fast. Nothing to complain about the laptop itself really aside from the even shallower keys in my opinion. ¯\\\_(ツ)\_/¯
